src/main/java/gameframework/motion/MoveStrategyConfigurableKeyboard.java

```diff
@@ -0,0 +1,117 @@
+package gameframework.motion;
+
+import java.awt.Point;
+import java.awt.event.KeyAdapter;
+import java.awt.event.KeyEvent;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * {@link MoveStrategy} which listens to the keyboard and answers new
+ * {@link SpeedVector speed vectors} based on what the user typed.
+ */
+public class MoveStrategyConfigurableKeyboard extends KeyAdapter implements MoveStrategy {
+	protected SpeedVector speedVector;
+	protected final Boolean alwaysMove;
+	protected Map<Integer, Point> directions;
+	protected Set<Integer> keyPressed;
+	protected boolean combineDirections;
+
+	public MoveStrategyConfigurableKeyboard() {
+		this(true);
+	}
+
+	public MoveStrategyConfigurableKeyboard(Boolean alwaysMove) {
+		this(alwaysMove, new SpeedVector(new Point(0, 0)));
+	}
+
+	public MoveStrategyConfigurableKeyboard(SpeedVector speedVector) {
+		this(true, speedVector);
+	}
+	
+	public MoveStrategyConfigurableKeyboard(Boolean alwaysMove, SpeedVector speedVector) {
+		this(alwaysMove, speedVector, false);
+	}
+
+	/**
+	 * @param alwaysMove is a boolean value that decide if a player moves continually or not. (True by default)
```

Comment:
your `by default` does not mean anything because in this constructor, a value has to be given: so, `true` has to be given explicitly.